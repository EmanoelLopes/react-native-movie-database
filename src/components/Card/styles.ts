import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  card: {
    borderRadius: 4,
    borderWidth: 1,
    borderColor: 'rgb(8, 28, 36)',
    flex: 1,
    height: 220,
    marginRight: 10,
    width: 160,
  },
  image: {
    alignItems: 'center',
    borderRadius: 4,
    height: '100%',
    justifyContent: 'flex-end',
    resizeMode: 'cover',
    width: '100%',
  },
  title: {
    textAlign: 'center',
    color: '#fff',
    fontWeight: '700',
    backgroundColor: 'rgba(8, 28, 36, 0.5)',
    width: '100%',
    padding: 10,
  },
});