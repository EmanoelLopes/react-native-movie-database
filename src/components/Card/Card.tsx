import React from 'react';
import { ImageBackground, View, Text } from 'react-native';
import { styles } from './styles';

type CardProps = {
  title: string;
  image: object;
};

export function Card({ title, image, ...rest }: CardProps) {
  return (
    <View style={styles.card} {...rest}>
      <ImageBackground source={image} style={styles.image}>
        <Text style={styles.title}>{title}</Text>
      </ImageBackground>
    </View>
  );
}
