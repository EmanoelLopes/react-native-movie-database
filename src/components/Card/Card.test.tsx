import React from 'react';
import { render } from '@testing-library/react-native';
import Card from '@components/Card';

describe('[Card]', () => {
  const props = {
    title: 'Mortal Kombat',
    image: {
      uri: 'https://image.tmdb.org/t/p/w500/9yBVqNruk6Ykrwc32qrK2TIE5xw.jpg',
    },
  };

  it('should render properly', () => {
    expect(render(<Card {...props} />)).toMatchSnapshot();
  });
});
