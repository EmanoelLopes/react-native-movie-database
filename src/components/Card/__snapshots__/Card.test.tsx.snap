// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Card] should render properly 1`] = `
<View
  style={
    Object {
      "borderColor": "rgb(8, 28, 36)",
      "borderRadius": 4,
      "borderWidth": 1,
      "flex": 1,
      "height": 220,
      "marginRight": 10,
      "width": 160,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 4,
        "height": "100%",
        "justifyContent": "flex-end",
        "resizeMode": "cover",
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://image.tmdb.org/t/p/w500/9yBVqNruk6Ykrwc32qrK2TIE5xw.jpg",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": "100%",
            "width": "100%",
          },
          undefined,
        ]
      }
    />
    <Text
      style={
        Object {
          "backgroundColor": "rgba(8, 28, 36, 0.5)",
          "color": "#fff",
          "fontWeight": "700",
          "padding": 10,
          "textAlign": "center",
          "width": "100%",
        }
      }
    >
      Mortal Kombat
    </Text>
  </View>
</View>
`;
